scilpy/image/operations.py

```diff
@@ -0,0 +1,471 @@
+        All values below the threshold will be set to one.
+        All values above the threshold will be set to zero.
+    lower_clip: IMG THRESHOLD
+        All values below the threshold will be set to threshold.
+    upper_clip: IMG THRESHOLD
+        All values above the threshold will be set to threshold.
+    absolute_value: IMG
+        All negative values will become positive.
+    round: IMG
+        Round all decimal values to the closest integer.
+    ceil: IMG
+        Ceil all decimal values to the next integer.
+    floor: IMG
+        Floor all decimal values to the previous integer.
+    normalize_sum: IMG
+        Normalize the image so the sum of all values is one.
+    normalize_max: IMG
+        Normalize the image so the maximum value is one.
+    convert: IMG
+        Perform no operation, but simply change the data type.
+    addition: IMGs
+        Add multiple images together.
+    subtraction: IMG_1 IMG_2
+        Subtract two images together.
+    multiplication: IMGs
+        Multiply multiple images together (danger of underflow and overflow)
+    division: IMG_1 IMG_2
+        Divide two images together (danger of underflow and overflow)
+    mean: IMGs
+        If a single 4D image is provided, average along the last dimension.
+    std: IMGs
+        Compute the standard deviation average of multiple images.
+        If a single 4D image is provided, compute the STD along the last
+        dimension.
+    union: IMGs
+        Binary operation to keep voxels that are in any file.
+    intersection: IMGs
+        Binary operation to keep the voxels that are present in all files.
+    difference: IMG_1 IMG_2
+        Binary operation to keep voxels from the first file that are not in
+        the second file.
+    invert: IMG
+        Binary operation to interchange 0 and 1 in a binary mask.
+    """
+
+def get_image_operation_doc():
+    return """dilation: IMG
+        Binary morphological operation to spatially expand the values of an
+        image to their neighbors.
+    erosion: IMG
+        Binary morphological operation to spatially shrink the values of an
+        image.
+    closing: IMG
+        Binary morphological operation, dilation followed by an erosion.
+    opening: IMG
+        Binary morphological operation, erosion followed by an dilation.
+    blur: IMG
+        Apply a gaussian blur to a single image.
+    """
+
+def is_float(value):
+    try:
+        float(value)
+        return True
+    except ValueError:
+        return False
+
+
+def find_array(input_list):
+    at_least_one_arr = False
+    ref_array = None
+    for i in input_list:
+        if isinstance(i, np.ndarray):
+            at_least_one_arr = True
+            ref_array = i
+            break
+
+    if not at_least_one_arr:
+        logging.error('At least one array is required in the input list.')
+        raise ValueError
+
+    return ref_array
+
+
+def lower_threshold(input_list):
+    if not len(input_list) == 2:
```

Comment:
I feel like we could cut everything with private functions like `_assert_len(input_list, length)`, `_assert_type(input[0], type)`, etc.